x-pack/legacy/plugins/ml/public/overview/overview_page.tsx

```diff
@@ -6,19 +6,28 @@
 
 import React, { Fragment, FC } from 'react';
 import { EuiFlexGroup, EuiPage, EuiPageBody } from '@elastic/eui';
+import { checkPermission } from '../privilege/check_privilege';
+import { mlNodesAvailable } from '../ml_nodes_check/check_ml_nodes';
 import { NavigationMenu } from '../components/navigation_menu';
 import { OverviewSideBar } from './components/sidebar';
 import { OverviewContent } from './components/content';
 
 export const OverviewPage: FC = () => {
+  const disableCreateAnomalyDetectionJob = !checkPermission('canCreateJob') || !mlNodesAvailable();
+  const disableCreateAnalyticsButton =
+    !checkPermission('canCreateDataFrameAnalytics') ||
+    !checkPermission('canStartStopDataFrameAnalytics');
   return (
```

Comment:
What do you think of moving the imports and check consts directly into the panel components? Then there would be no need to pass them down 2 levels via props.

Reply:
Keeping them in the parent component makes it easier to share the permissions check since it has to be called from the panels and the sidebar component. I thought it might be preferable to calling it in 3 different places. Happy to make the change if it seems cleaner.

Reply:
Sorry I missed that the same check is used in Panel + Sidebar, happy to leave as is!